client/extensions/woocommerce/app/dashboard/index.js

```diff
@@ -149,8 +149,13 @@ class Dashboard extends Component {
 			return <RequiredPagesSetupView site={ selectedSite } />;
 		}
 
-		if ( ! setStoreAddressDuringInitialSetup && ! hasProducts ) {
-			return <PreSetupView siteId={ selectedSite.ID } />;
+		if ( ! setStoreAddressDuringInitialSetup ) {
```

Comment:
Why was `! hasProducts` removed?

Reply:
Because with it present, we get the blank experience - this way we still have the user explicitly verify their address (once) but if they have products, we will no longer override their dimension, currency, weight settings